tests/BladeTest.php

```diff
@@ -184,6 +194,22 @@ public function the_hasallroles_directive_will_evaluate_true_when_the_logged_in_
         $this->assertEquals('does have all of the given roles', $this->renderView('guardHasAllRoles', compact('roles', 'guard')));
     }
 
+    /** @test */
+    public function the_hasallroles_directive_will_evaluate_true_when_the_logged_in_user_does_have_all_required_roles_pipe_for_the_given_guard()
+    {
+        $guard = 'admin';
+
+        $admin = $this->getSuperAdmin();
+
+        $admin->assignRole('moderator');
+
+        $this->refreshTestAdmin();
+
+        auth('admin')->setUser($admin);
+
+        $this->assertEquals('does have all of the given roles', $this->renderView('guardHasAllRolesPipe', compact('guard')));
```

Comment:
Could you add a tests the checks if `does not have all of the given roles` when the user does not have all the given roles?